MARIE_AND_BERT/readme.md

```diff
@@ -0,0 +1,141 @@
+created. To create or amend the dataset, see [readme.md for dataset creation](./KGToolbox/readme.md)
+
+
+## Training
+Multiple models are trained in this system: 
+1. Knowledge Graph Embedding
+2. Relation Prediction
+3. Score Alignment
+4. Entity Linking 
+
+For details on training the models, see [readme.md for embedding and model training](./Training/readme.md) and the [readme.md for entity linking training](./Training/EntityLinking/EL_training.md)
+ 
+  
+## Running 
+
+Install `Python 3.8` and Run `pip install requirements.txt`. It is recommended 
+to use conda for creating a virtual environment. 
+
+### Required files: 
+
+1. Create a `DATA` folder under `MARIE_AND_BERT`
+2. Download CrossGraph.zip from [CrossGraph.zip](http://159.223.42.53:8080/CrossGraph.zip), unzip under `DATA`
+3. Download Dictionaries.zip from [Dictionaries.zip](http://159.223.42.53:8080/Dictionaries.zip), unzip under `DATA`
+4. For faster setup, download the `bert_pretrained.zip` from [bert_pretrained.zip](http://159.223.42.53:8080/bert_pretrained.zip)
+, unzip under `DATA`. Otherwise upon the first run, the scripts will download the BERT pretrained model from hugging face.
+5. Download EntityLinking.zip from [EntityLinking.zip](http://159.223.42.53:8080/EntityLinking.zip), unzip under `DATA`
+6. Download STOUT V2 model [STOUT V2 model](http://159.223.42.53:8080/models.zip) for Linux system, unzip into `root/.data/STOUT-V2/`, for Windows system,
+7. Download `label_dict.js` from [label_dict.js] from (http://159.223.42.53:8080/label_dict.js) and put the file in `MARIE_ANB_BERT/static/js`.
+8. Download the required NLTK datasets by running
+```python
+import nlkt
+nltk.download('all')
+```
+ 
+To start the system, run `python main.py`.
+
+For debugging purpose, in `main.py` please comment the line
+
+```python
+    app.run(host='0.0.0.0', debug=False, port=5003, threaded=False, processes=1)
+```
+
+and uncomment the line
+```python
+    # app.run(host='0.0.0.0', debug=True, port=5003)
+```
+
+### Other Services
+To run the full functions of the Marie system, three other systems are required:
+
+1. The LDF server. See [LDF server readme](../JPS_LDF/README.md) to run it. 
+2. The semantic agents. See [PCE agent readme](../Agents/PCEAgent/README.md) 
+[Thermal agent readme](../Agents/STDCThermoAgent/README.md) to create docker containers running them. 
+
+
+## Docker Deployment
+
+For local deployment，please use `Dockerfile_local`. 
+1. run `docker build  --no-cache -t marie .d -f Dockerfile_local .` to build the image
+2. run ` docker run -p 5003:80 -d marie:latest`. 
+3. The Marie web-interface will then be available at `http://localhost:5003` or `http://127.0.0.1:5003`
+
+For deployment on Linux server from scratch:
+
+1. Create a folder in the server, assume it is `/home/user1/Marie`
+2. Clone the GitHub repository by `git clone https://github.com/cambridge-cares/TheWorldAvatar`, Assume the users cloned the repository in `/home/user1/Marie`
```

Comment:
I wonder if this is unnecessarily complicated. Anyone who will read this is essentially guaranteed to already have a local clone in their user space. Why create another one? The `KG` folder referred to in step 4 could easily sit in `MARIE_AND_BERT/DATA/KG`, or even `MARIE_AND_BERT/KG` if you prefer (just make sure you `.gitignore` it).